### PR TITLE
All dep groups set to optional

### PR DESCRIPTION
**Pull request by dragomirp**
# Issue
* Poetry groups get installed in tox environments that don't need them

# Solution
* Mark all dependency groups as optional, so they get installed only when pulled explicitly

# Context
* Also fixed some typos and bumped dependency versions

# Testing

# Release Notes
Housekeeping

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # See LICENSE file for licensing details.
 
 [tool.poetry]
-name = "postgesql-k8s"
+name = "postgresql-k8s"
```
Comment (taurus-forever) [inline pyproject.toml:5]: o_0 how it worked?

**Review by taurus-forever — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (dragomirp) [reply pyproject.toml:5]: This is just poetry metadata, it requires it but it's not really used for anything.

**Comment by dragomirp** [comment]
New libjuju is out (2.9.42.1). Updating as part of this PR.

**Comment by taurus-forever** [comment]
> 2.9.42.1

The [changelog](https://github.com/juju/python-libjuju/compare/2.9.38...2.9.42.1) says: no changes expected.

**Review by marceloneppel — APPROVED** [review]

**Pull request merged**